src/crypto.rs

```diff
@@ -168,6 +169,70 @@ pub(crate) fn asym_verify(
     Ok(verifier.verify(signature.as_bytes())?)
 }
 
+pub(crate) fn rsa_oaep_decrypt(
+    priv_key: &PKey<Private>,
+    data: &[u8],
+) -> Result<Vec<u8>> {
+    let mut decrypter = Decrypter::new(priv_key)?;
+
+    decrypter.set_rsa_padding(Padding::PKCS1_OAEP)?;
+    decrypter.set_rsa_mgf1_md(MessageDigest::sha1())?;
+    decrypter.set_rsa_oaep_md(MessageDigest::sha1())?;
+
+    // Create an output buffer
+    let buffer_len = decrypter.decrypt_len(data)?;
+    let mut decrypted = vec![0; buffer_len];
+
+    // Decrypt and truncate the buffer
+    let decrypted_len = decrypter.decrypt(data, &mut decrypted)?;
+    decrypted.truncate(decrypted_len);
+
+    Ok(decrypted)
+}
+
+pub(crate) fn compute_hmac(key: &[u8], data: &[u8]) -> Result<Vec<u8>> {
+    let pkey = PKey::hmac(key)?;
+    let mut signer = Signer::new(MessageDigest::sha384(), &pkey)?;
```

Comment:
I should have done this in the original version, but maybe it would be good to document why we are using these algos (they are expected on the Python side) - with refs to the Python files / line numbers. I can also do this.